src/components/Comment/index.tsx

```diff
@@ -13,10 +13,42 @@ import {
 } from '../../actions/comments';
 import { LayoutController } from '../../utils/layout';
 import { getNextReplyId } from '../../utils/sequences';
-import CommentReplyComponent from '../CommentReply';
+import CommentReplyComponent, { addRemoteWarningMessage as addRemoteReplyWarningMessage } from '../CommentReply';
+import CommentMessage from '../CommentMessage';
 import type { TranslatableStrings } from '../../main';
 
-async function saveComment(comment: Comment, store: Store) {
+function addRemoteWarningMessage(
+  comment: Comment,
+  message: string,
+  store: Store,
+  addIfLocal = false,
+  onFinish?: () => void
+) {
+  if (addIfLocal || comment.remoteId !== null) {
+    // Only add the warning message if the comment actually exists in the db - editing a comment you've created
+    // locally, for instance, requires no such warning
+    store.dispatch(
+      updateComment(comment.localId, {
+        message: message,
+      })
+    );
+    window.setTimeout(() => {
+      store.dispatch(
+        updateComment(comment.localId, {
+          message: '',
+        })
+      );
+      if (onFinish !== undefined) {
+        onFinish();
+      }
+    },
+    1500);
+  } else if (onFinish !== undefined) {
+    onFinish();
+  }
+}
+
+async function saveComment(comment: Comment, store: Store, strings: TranslatableStrings, forceWarning = false) {
```

Comment:
I’m not sure how much it matters but feels a bit odd to me to have to pass this `strings` object around. I feel like this might work better by having a predefined set of string IDs, and then whichever component does the rendering would be the only one that needs access to the human-friendly labels?

Something like:

```ts
const translatableStrings = {
  SAVE: 'SAVE';
  SAVING: 'SAVING';
  CANCEL: 'CANCEL';
  DELETE: 'DELETE';
  DELETING: 'DELETING';
  SHOW_COMMENTS: 'SHOW_COMMENTS';
  EDIT: 'EDIT';
  REPLY: 'REPLY';
  RESOLVE: 'RESOLVE';
  RETRY: 'RETRY';
  DELETE_ERROR: 'DELETE_ERROR';
  CONFIRM_DELETE_COMMENT: 'CONFIRM_DELETE_COMMENT';
  SAVE_ERROR: 'SAVE_ERROR';
  SAVE_COMMENT_WARNING: 'SAVE_COMMENT_WARNING';
} as const;

export type TranslatableStrings = keyof typeof translatableStrings;
```

Or a bit more idiomatic TypeScript,

```ts

const translatableStrings = [
  'SAVE',
  'SAVING',
  'CANCEL',
  'DELETE',
  'DELETING',
  'SHOW_COMMENTS',
  'EDIT',
  'REPLY',
  'RESOLVE',
  'RETRY',
  'DELETE_ERROR',
  'CONFIRM_DELETE_COMMENT',
  'SAVE_ERROR',
  'SAVE_COMMENT_WARNING',
 ] as const;

export type TranslatableStrings = typeof translatableStrings[number];
```

Or an enum, but that’s a bit too idiomatic for my taste.